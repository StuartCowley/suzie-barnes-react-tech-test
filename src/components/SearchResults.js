import React from "react";
import "../styles/searchresults.css";
import PropTypes from "prop-types"

const SearchResults = ({ results }) => {

    if (!results.length) {
        return <p>No results</p>
    } else {
        return (
            <div className="search-results">
                {
                    results.map((image, index) => (
                        <img
                            key={index}
                            className="search-results__images"
                            src={image}
                            alt="search-results__image"
                            data-testid="search-results-test"
                        />
                    ))
                }
            </div>
        )
    }
}
SearchResults.defaultProps = {
    results: ""
}

SearchResults.propTypes = {
    results: PropTypes.array.isRequired,
}


export default SearchResults;
