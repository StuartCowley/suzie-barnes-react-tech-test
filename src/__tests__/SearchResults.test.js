import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
    const validProps = {
        results: [
            {
                key: [123],
                className: "search-results__images",
                src: "StubImage1",
                alt: "search-results__image",
            },

            {
                key: [124],
                className: "search-results__images",
                src: "StubImage2",
                alt: "search-results__image",
            },

            {
                key: [125],
                className: "search-results__images",
                src: "StubImage3",
                alt: "search-results__image",
            }
        ]
    };
    const { asFragment } = render(
        <SearchResults
            results={validProps.results}
        />);

    render(
        <SearchResults
            results={validProps.results}
        />
    );
    it("renders correctly", () => {
        expect(asFragment()).toMatchSnapshot()
    })

    it("renders the correct number of images", () => {
        render(
            <SearchResults
                results={validProps.results}
            />
        );
        expect(screen.getAllByTestId("search-results-test")).toHaveLength(3)
    })

    it("loads the heading Image Results", () => {
        render(
            <SearchResults
                results={validProps.results}
            />
        );
        expect(screen.getByText("Image Results:")).toBeInstanceOf(HTMLHeadingElement)
    })

    it("loads No results if no results found", () => {
        render(
            <SearchResults
                results={[]}
            />
        );
        expect(screen.getByText("No results")).toBeInstanceOf(HTMLParagraphElement);
    })
})