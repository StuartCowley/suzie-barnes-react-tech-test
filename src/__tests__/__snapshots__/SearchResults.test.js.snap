// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <h1>
      Image Results:
    </h1>
    <img
      alt="search-results__image"
      class="search-results__images"
      data-testid="search-results-test"
      src="[object Object]"
    />
    <img
      alt="search-results__image"
      class="search-results__images"
      data-testid="search-results-test"
      src="[object Object]"
    />
    <img
      alt="search-results__image"
      class="search-results__images"
      data-testid="search-results-test"
      src="[object Object]"
    />
  </div>
</DocumentFragment>
`;
